apps/waterfall/waterfall/src/app/dashboard/title/edit/edit.component.ts

```diff
@@ -0,0 +1,92 @@
+// Angular
+import { ActivatedRoute } from '@angular/router';
+import { FormControl, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
+import { BehaviorSubject, Observable, combineLatest, map, startWith } from 'rxjs';
+
+// Blockframes
+import { RightholderRole } from '@blockframes/model';
+import { MovieService } from '@blockframes/movie/service';
+import { MovieForm } from '@blockframes/movie/form/movie.form';
+import { OrganizationService } from '@blockframes/organization/service';
+import { FileUploaderService } from '@blockframes/media/file-uploader.service';
+import { WaterfallPermissionsService } from '@blockframes/waterfall/permissions.service';
+
+
+@Component({
+  selector: 'waterfall-title-edit',
+  templateUrl: './edit.component.html',
+  styleUrls: ['./edit.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class EditComponent implements OnInit {
+
+  movieForm = new MovieForm({ directors: [{ firstName: '', lastName: '' }] });
+
+  waterfallForm = new FormControl<RightholderRole[]>(undefined, [Validators.required]);
+
+  movieId = '';
+
+  invalid$: Observable<boolean>;
+
+  loading$ = new BehaviorSubject(true);
+  updating$ = new BehaviorSubject(false);
+
+  constructor(
+    private snackBar: MatSnackBar,
+    private route: ActivatedRoute,
+    private movieService: MovieService,
+    private orgService: OrganizationService,
+    private uploadService: FileUploaderService,
+    private permissionsService: WaterfallPermissionsService,
+  ) { }
+
+  async ngOnInit() {
+    this.movieId = this.route.snapshot.params.movieId;
+    const [ movie ] = await this.movieService.getValue([this.movieId]);
+    this.movieForm.patchValue(movie);
+
+    const permissions = await this.permissionsService.getValue(this.orgService.org.id, { waterfallId: this.movieId });
+    this.waterfallForm.patchValue(permissions.roles as any);
```

Comment:
you don"t need to cast `as any` and you can pass directly a string to `getValue()`
```suggestion
    const [movie, permissions] = await Promise.all([
      this.movieService.getValue(this.movieId),
      this.permissionsService.getValue(this.orgService.org.id, { waterfallId: this.movieId })
    ]);
    
    this.movieForm.patchValue(movie);
    this.waterfallForm.patchValue(permissions.roles);
```